### PR TITLE
SDFG Editing

### PR DESCRIPTION
**Pull request by phschaad**
This continues work started by @tobiasae in #18.

- [x] Add library nodes
- [x] Add states
- [x] Add general nodes
- [x] Add map scopes
- [x] Add consume scopes
- [x] Edges
- [x] ~Add connectors~ The current interface is enough for the time being, other connectors should be automatically added by adding edges in Phase 2
- [x] Delete elements / edges

### CODE REVIEW

**Comment by phschaad** [comment]
This requires SDFG element property editing to be useful, tracking #40.

File: backend/run_dace.py [hunk 1]
```diff
@@ -659,6 +674,223 @@ def compile_sdfg(path, suppress_instrumentation=False):
         'filename': compiled_sdfg.filename,
     }
 
+def recursively_find_graph(graph, graph_id, ns_node = None):
```
Comment (tbennun) [inline backend/run_dace.py:677]: consider splitting the file

**Review by tbennun — APPROVED** [review]

**Comment by tbennun** [comment]
@phschaad remember to update submodule after webclient is merged.
Seems like there is a merge conflict here too

File: media/components/sdfv/index.html [hunk 2]
```diff
@@ -34,30 +34,137 @@
...
-                <span>
-                    Refresh SDFG
-                </span>
+        <div id="search-bar-wrapper">
+            <div id="search-bar" class="container">
+                <div id="search-container">
+                    <input id="search" type="text" placeholder="Search the graph">
+                </div>
+                <button id="search-case-sensitive-btn" checked="false"
+                        title="Search case sensitively"
```
Comment (tbennun) [inline media/components/sdfv/index.html:43]:
```suggestion
                        title="Case sensitive"
```

**Review by tbennun — APPROVED** [review]
looks much better now. One minor comment

**Pull request merged**